File: Lua/twosum.lua
local nums = { 2, 4, 5, 15 }
local target = 9

local function twosum(nums, target)
    for i, num in ipairs(nums) do
        for j , num2 in ipairs(nums) do
            if i ~= j and num + num2 == target then
                return i, j
            end
        end
    end
end

local result1, result2 = twosum(nums, target)
print(string.format("nums = %s", table.concat(nums, ", ")))
print(string.format("target = %d", target))
print(string.format("result is:\n %d %d", result1, result2))

print("Also note that a feature of LUA is that indexing starts at 1, not 0.\n")
print(string.format("Therefore, the result for the function with traditional indexing would be:\n %d %d", result1 - 1 , result2 - 1))

File: Ruby/twosum.rb
nums = [1, 3, 5, 7, 9]
target = 12

def two_sum(nums, target)
    for num, i in nums.each_with_index
        for num2, j in nums.each_with_index
            if j != i and num + num2 == target
                return [i, j]
            end
            if j and i == nums.length - 1
                return nil
            end
        end
    end
end

puts two_sum(nums, target)

#tests for two_sum
puts two_sum([1, 2, 3], 4) == [0, 2]
puts two_sum([1, 2, 3], 5) == [1, 2]
puts two_sum([1, 2, 3], 6) == nil
puts two_sum([1, 2, 3], 7) == nil
puts two_sum([1, 2, 3], 8) == nil
puts two_sum([1, 2, 3], 9) == nil
puts two_sum([1, 2, 3], 10) == nil